PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -846,77 +846,140 @@ void DefaultBinning()
-  if (lFixedLength_eta_bins.size() != 3) {
-    LOGF(fatal, "in function \033[1;31m%s at line %d => The array cfFixedLength_eta_bins must have 3 entries: {nBins, min, max} \n \033[0m", __FUNCTION__, __LINE__);
-  }
-  res.fResultsProFixedLengthBins[AFO_ETA][0] = lFixedLength_eta_bins[0];
-  res.fResultsProFixedLengthBins[AFO_ETA][1] = lFixedLength_eta_bins[1];
-  res.fResultsProFixedLengthBins[AFO_ETA][2] = lFixedLength_eta_bins[2];
+  this->InitializeFixedLengthBins(AFO_ETA);
 
   // e) Variable-length binning set via MuPa-Configurables.h:
   // *) Variable-length binning vs. multiplicity:
   if (cf_res.cfUseVariableLength_mult_bins) {
-    res.fUseResultsProVariableLengthBins[AFO_MULTIPLICITY] = kTRUE;
-    res.fResultsProVariableLengthBinsString[AFO_MULTIPLICITY] = cf_res.cfVariableLength_mult_bins;
-    this->CastStringIntoArray(AFO_MULTIPLICITY);
+    this->InitializeVariableLengthBins(AFO_MULTIPLICITY);
   }
   // *) Variable-length binning vs. centrality:
   if (cf_res.cfUseVariableLength_cent_bins) {
-    res.fUseResultsProVariableLengthBins[AFO_CENTRALITY] = kTRUE;
-    res.fResultsProVariableLengthBinsString[AFO_CENTRALITY] = cf_res.cfVariableLength_cent_bins;
-    this->CastStringIntoArray(AFO_CENTRALITY);
+    this->InitializeVariableLengthBins(AFO_CENTRALITY);
   }
   // *) Variable-length binning vs. pt:
   if (cf_res.cfUseVariableLength_pt_bins) {
-    res.fUseResultsProVariableLengthBins[AFO_PT] = kTRUE;
-    res.fResultsProVariableLengthBinsString[AFO_PT] = cf_res.cfVariableLength_pt_bins;
-    this->CastStringIntoArray(AFO_PT);
+    this->InitializeVariableLengthBins(AFO_PT);
   }
   // *) Variable-length binning vs. eta:
   if (cf_res.cfUseVariableLength_eta_bins) {
-    res.fUseResultsProVariableLengthBins[AFO_ETA] = kTRUE;
-    res.fResultsProVariableLengthBinsString[AFO_ETA] = cf_res.cfVariableLength_eta_bins;
-    this->CastStringIntoArray(AFO_ETA);
+    this->InitializeVariableLengthBins(AFO_ETA);
   }
 
 } // void DefaultBinning()
 
 //============================================================
 
+void InitializeFixedLengthBins(eAsFunctionOf AFO)
+{
+  // This is a helper function to suppress code bloat in DefaultBinning().
+  // In merely initalize res.fResultsProFixedLengthBins[...] from corresponding configurables + a few other minor thingies.
+
+  if (tc.fVerbose) {
+    LOGF(info, "\033[1;32m%s\033[0m", __FUNCTION__);
+  }
+
+  // Common local vector for all fixed-length bins:
+  vector<float> lFixedLength_bins;
+
+  switch (AFO) {
+    case AFO_MULTIPLICITY:
+      lFixedLength_bins = (vector<float>)cf_res.cfFixedLength_mult_bins;
+      break;
+    case AFO_CENTRALITY:
+      lFixedLength_bins = (vector<float>)cf_res.cfFixedLength_cent_bins;
+      break;
+    case AFO_PT:
+      lFixedLength_bins = (vector<float>)cf_res.cfFixedLength_pt_bins;
+      break;
+    case AFO_ETA:
+      lFixedLength_bins = (vector<float>)cf_res.cfFixedLength_eta_bins;
+      break;
```

Comment:
Probably using 
```
cf_res.cfFixedLength_mult_bins.value
```
will make the cast not necessary

Consider it for the next iteration

Reply:
Done! I will be shipped with the next commit.